src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/NodeMonitoringConnectionPlugin.java

```diff
@@ -316,9 +320,13 @@ private void generateNodeKeys(Connection connection) {
         }
       }
     } catch (SQLException sqlException) {
```

Comment:
Would it be easier to catch SQLSyntaxException instead of performing complex checks to detect communication excpetions? The only reason to swallow exception here is a case when DB doesn't support extended Aurora functions. Vanilla Mysql doesn't have them so exception (SQLSyntaxException) should be ignored. All other exceptions should be popped up. Such solution would (maybe) simplify the code since connectionMethodAnalyzer might not be needed.

Reply:
Sounds good